src/core/libmaven/jsonReports.cpp

```diff
@@ -1,286 +1,674 @@
+                               _mavenParameters->eicType,
+                               _mavenParameters->filterline);
+    }
+
+    //TODO: for MS1 we've already limited RT range, but for MS/MS the entire RT range of the SRM will be output
+    //either check here or edit getEIC functionality
+    if(eic) {
+        int N = eic->rt.size();
+        myfile << ",\n" << "\"eic\": {" ;
+        myfile << "\"rt\": [";
+
+        for(int i = 0; i < N; i++){
+            if ( eic->rt[i] > 0) {
+                myfile << eic->rt[i];
+                if (i < N - 1) myfile << ",";
+            }
+        }
+            
+        myfile << "],\n" ; //rt
+        myfile << "\"intensity\": [";
+        for(int i= 0; i < N; i++){
+            if ( eic->rt[i] > 0) {
+                myfile << eic->intensity[i];
+                if (i < N - 1) myfile << ",";
+            }
+        }
+            
+        myfile << "]" ; //intensity
+        myfile << "\n}" ;//eic
+        myfile << "\n}" ; //peak
+        delete(eic);
+    }
+   
+}
+
+
+
+void JSONReports::save(string filename, vector<PeakGroup> allgroups, vector<mzSample*> samples)
+{
+    ofstream myfile(filename.c_str());
+    myfile << setprecision(10);
+
+    myfile << "{\"groups\": [" <<endl;
+
+    int groupId = 0;
+    int metaGroupId = 0;
+   
+    for(size_t i=0; i < allgroups.size() ; i++ ) {
+        PeakGroup& grp = allgroups[i];
+
+        //if compound is unknown, output only the unlabeled form information
+        if(grp.getCompound() == NULL || grp.childCount() == 0) {
+            grp.groupId = ++groupId;
+            grp.metaGroupId = ++metaGroupId;
+            if(groupId > 1) myfile << "\n,";
+
+            _writeGroup(grp,myfile);  
+            if(grp.hasCompoundLink())
+                _writeCompoundLink(grp, myfile);
+            _writePeak(grp, myfile, samples);
+
+        } else {
+            //output all relevant isotope info otherwise
+            //does this work? is children[0] always the same as grp (parent)?
+            grp.metaGroupId = ++ metaGroupId;
+            for (unsigned int k = 0; k < grp.children.size(); k++) {
+                grp.children[k].metaGroupId = grp.metaGroupId;
+                grp.children[k].groupId = ++groupId;
+                if(groupId > 1) myfile << "\n,";
+            
+                _writeGroup(grp.children[k], myfile);
+                if( grp.children[k].hasCompoundLink() )
+                    _writeCompoundLink(grp, myfile);
+                _writePeak(grp.children[k], myfile, samples);
+            }
+        }
+    }
+    myfile << "]}"; //groups
+    myfile.close();
+}
+
+string JSONReports::_sanitizeJSONstring(string s)
+{
+    boost::replace_all(s, "\"","\\\"");
+    s = "\""+s+"\"";    //quote the whole string
+    return s;
+}
+
+
+class JsonReportsFixture{
+
+    protected:
+            vector<mzSample*> samples;
+            vector <PeakGroup> allgroups;
+            MavenParameters* mavenparameters;

```

Comment:
I am curious why you made these variables protected and not private.

Reply:
Making them private in fixture would limit their access in the test_case. Because we needed maven parameters and other variables to make json object and  call the corresponding function. It was a necessity to make them protected and not private.

Reply:
Since the test cases only need to access these structures, they can simply be functions:
```C++
vector<mzSample*> samples();
vector <PeakGroup> allgroups();
MavenParameters* mavenparameters();
```